dmi_id_dbus_service/src/main.rs

```diff
@@ -0,0 +1,76 @@
+/*
+ * This file is part of Edgehog.
+ *
+ * Copyright 2022 SECO Mind Srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use std::{
+    error::Error
+};
+use zbus::{ConnectionBuilder, dbus_interface};
+use std::fs;
+
+struct Device {
+    id: String,
+}
+
+#[dbus_interface(name = "org.secomind.Device1")]
+impl Device {
+    #[dbus_interface(property)]
+    async fn device_id(&self) -> &str {
+        &self.id
+    }
+}
+
+// Get hardware id from board serial or from product uuid provided by the BIOS (standard SMBIOS/DMI).
+fn get_hardware_id() -> Option<String> {
+    let mut data = fs::read_to_string("/sys/class/dmi/id/board_serial")
+        .expect("Unable to read file")
+        .trim()
+        .to_string();
+    if data != "0" {
+        return Some(data);
+    } else {
+        data = fs::read_to_string("/sys/class/dmi/id/product_uuid")
+            .expect("Unable to read file")
+            .trim()
+            .to_string();
+        if data != "" {
+            return Some(data);
+        }
+    }
+    None
+}
+
+// Simple DBUS service that retrieves a machine specific id and publish it on a system channel
+#[async_std::main]
```

Comment:
Is there a reason to use async_std? We are using tokio for everything else

Reply:
You are right, I forgot about it. Fixed